core/xml.js

```diff
@@ -332,6 +338,71 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
 };
 
 /**
+ * Decode an XML DOM and create blocks on the workspace.
+ * Append the new blocks at the bottom of the existing blocks
+ * @param {!Element} xml XML DOM.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.appendDomToWorkspace = function(xml, workspace) {
+  
+  // find bottom position and alignment of the current blocks  
+  //var blocks = workspace.getAllBlocks();
+  var blocks = workspace.getTopBlocks();
+  var offsetY = 0; // container for the y of the bottom block
+  var offsetX = 0; 
+  var blocksCount = blocks.length;
+  if (blocksCount > 0) {
+	// initialize the position of the corners using position 
+	// of the first block
+	var bHW = (("height" in blocks[0])? blocks[0].getHeightWidth() : {height: 0, width: 0});
```

Comment:
I suppose you assume that all blocks have been loaded in the workspace so we can use getBlocksBoundingBox() and getBoundingRectangle().
In the current situation, the new blocks are not affected by the TAB since this is not included in the xml position. From what I understand from @picklesrus, the way I suggest:
var savetab = Blockly.BlockSvg.TAB_WIDTH;
Blockly.BlockSvg.TAB_WIDTH = 0;
var bbox = workspace.getBlocksBoundingBox();
Blockly.BlockSvg.TAB_WIDTH = savetab;  

may only be problematic if the user is doing something else requesting TAB_WIDTH at the time he is loading a file  with the domToWorkspace function. Doing this will take care of case 5.